mne/viz/topomap.py

```diff
@@ -58,12 +60,17 @@ def _prepare_topo_plot(inst, ch_type, layout):
             raise ValueError("No channels of type %r" % ch_type)
 
         if layout is None:
-            from ..channels.layout import _find_topomap_coords
             pos = _find_topomap_coords(info, picks)
         else:
             names = [n.upper() for n in layout.names]
-            pos = [layout.pos[names.index(info['ch_names'][k].upper())]
-                   for k in picks]
+            try:
+                pos = [layout.pos[names.index(info['ch_names'][k].upper())]
+                       for k in picks]
+            except ValueError:  # special case, where pos info not in layout
+                logger.warning('Failed to locate %s channel positions from '
+                               'layout. Inferring channel positions from '
+                               'data.' % ch_type)
```

Comment:
by using an exceptions you make the code harder to read as it's unclear when reading that error you are anticipating.
Something like this is more readable I think:

```
pos = []
for k in picks:
    this_name = info['ch_names'][k].upper()
    if this_name not in names:
        logger.warning(...)
        break
    else:
        pos.append(layout.pos[names.index(this_name)])
```